### ISSUE TITLE
Reduce Generic User Sign Up Fields

### ISSUE THREAD

**Issue by joshdanielson** [loconomics/loconomics#312]
## Child Stories
- As  [Sara the Service Professional](https://github.com/loconomics/loconomics/blob/master/docs/Personas/Sara%20the%20Service%20Professional.md) I want to **enter** only the *minimum information required* to sign up as I don't know yet why you need it.

### Test Cases 
1) Make sure we have the user's email address.
2) Make sure they create a password.
3) Make sure to capture the country.
4) Make sure they've agreed to the terms and conditions and privacy policy.
6) Make sure we know if they're a client or service professional.
7) Make sure we know the marketing source they came from.
8) Make sure the user can connect with facebook with one click
9) Make sure we store information we get from facebook for later forms, e.g. first name, last name
10) Make sure requirements later are validated if they were originally required in the sign-up form, e.g. first name, last name, phone number, postal code.

### Front-end
- [x] /signup : 
   - Reduce fields to:
     - email address
     - password
     - country (preset this to US)
     - checkbox to agree to terms
     - pass in 14 days as trial end date

### JS
- 
### API
- 
### Database
- 
### Braintree
-

**Comment by situatedbit** [comment]
5

**Comment by IagoSRL** [comment]
estimate:3

**Comment by joshdanielson** [comment]
3

**Comment by IagoSRL** [comment]
@joshdanielson At sign-up we say "Start your free 30-day trial" do you want to change that?

**Comment by IagoSRL** [comment]
I just read the 14 days note, I suppose just change the day in the language but keep it displayed as now, right?

**Comment by joshdanielson** [comment]
@iagosrl with this first listing free I think we should go with that instead of the trial when initially signing up. When we do the plans, we can message the 14 days. Look at the landing page copy.

**Comment by IagoSRL** [comment]
OK, text replaced with one from welcome/messageTherapists

**Comment by IagoSRL** [comment]
@joshdanielson I think this is done or nearly done. I removed fields visually while keeping them in the viewmodel, changes REST API and added an option at the signup template/viewmodel to hide country when not wanted; I commented this to @situatedbit already so can look the code and use at issue #366 that depends on this.
I uploaded my changes to *dev* so @joshdanielson you can see and test there. Let me know any comment or change wanted; if ready we can close and merge to master.

**Comment by joshdanielson** [comment]
@IagoSRL just tested this with some input from @situatedbit. The only problems were with the facebook login. It should be one click if they "sign up with facebook". If we can't get the country from facebook, let's just default it to US for all facebook sign ups for now. And I tried changing my email address after clicking sign up with facebook and then tried connecting with facebook again and it said "incorrect user". We should just assume they receive emails with the address connected with their facebook account. 

So it should go straight to onboarding if they click "sign up with facebook"
Does that make sense?

**Comment by IagoSRL** [comment]
You are right. I know this behavior but in local couldn't test it, will change

**Comment by IagoSRL** [comment]
Change done and uploaded to dev!!

**Comment by joshdanielson** [comment]
@IagoSRL working! It does have a little lag and shows that screen for half a second or so. Not sure how to get rid of that. Not a huge deal, though!

**Comment by situatedbit** [comment]
@IagoSRL Josh and I talked about this, and we think clients should include name and phone during signup. Unlike service professionals, they do not have another chance during onboarding to include these details. Here's my suggestion:

1. Can you include an ``options`` parameter to the ``SignupVM`` function that allows the caller to specify if they want to hide these fields? I'd like to use an ``options`` parameter later anyway.
2. The signup activity should include these fields, but learnMoreProfessionals and landingPages will not include these fields. (But in both of these cases it can be handled by the signup view model).
3. It's not clear to me yet how to handle validation in this case. If these fields are visible on the form, then they should be required. Somehow the server-side validation is going to have to know whether these are required.

Once we have this settled, we can merge our work into #226.

1. I will merge my work (still ongoing) from #366 into #226. 
2. You can merge #312 into master, then merge master into #226. There will be a couple merge conflicts around the country field:
  - I am now using the ko ``option`` binding. Without this, KO was not picking up on the default value we set with the ``selected`` attribute in the ``country-options`` template.
  - I also set the default country in code. We'll have to resolve this discrepancy in learnMoreProfessionals
3. You can start working on #367 or #331 with a branch off of #226. 
4. I will finish up #366 and merge into #226 

This is confusing.

**Comment by situatedbit** [comment]
I have merged my current work from #366 into the #226 feature branch.

**Comment by IagoSRL** [comment]
Requested changes done, #312 merged into master and master into #226, conflicts fixed.

Important notes:
- I did **not** include an options in SignupVM, but I did the change in the same way as before we implemented this differences:
  - based on the profileType value (defaults to client), we display or hide fields
  - at server, we use that same value to conditionally set-up validations (we had this before too)
- Previous means no need to set-up any special to hide client-only field
  - the same previous setting to preset the profile type does the work
  - I just changed how that value is set by using a new enum: SignupVM.profileType.serviceProfessional rather than literal 'service-professional', but it gets that same internal value same used in the URL for signup activity.
- Current way to do what the proposal *options* would allow is set value directly on the observables at the SignupVM instance (as we are already doing); this has the benefit that can change the value after initialization, something needed at signup activity.
  - I was thinking in wrap the signup in a component, with parameters to set that settings and handlers for events, so could be easily used at different landing pages html without touching the shared js.
  - I see your are thinking in share the html and keep the different set-ups on js, is the opposite of the initial idea and means Josh will have to touch js rather than html for new landing pages.

I will work after lunch in the remaining issues.

**Comment by situatedbit** [comment]
> I was thinking in wrap the signup in a component, with parameters to set that settings and handlers for events, so could be easily used at different landing pages html without touching the shared js.

Do you mean creating a knockout component for signup? I think this could work. And yes, I'd rather Josh modify markup than JS :)

**Comment by IagoSRL** [comment]
@joshdanielson I did a minor change where Facebook signup is working to hide other fields and communicate state through the button text 'Signing up with Facebook...', to prevent the undesired effect you commented.

I merged this issue branch (is312-reduce-signup) into the landing pages branch #226 (is226-create-profession-specific-landing-pages). Same as with #366, I think we must already stop committing and close this is312; any last minute fix to this must go to is226 directly from now, preventing conflicts and simplifying bugfixes.

Anyway, I keep this open for some minutes at QA as a reminder to test it and comment here any bug.
